### ISSUE TITLE
Laptop freezes completely when loading model

### ISSUE THREAD

**Issue by Lion-kh** [lllyasviel/Fooocus#1137]
I am very grateful for the amazing image creation tool! Please tell me, I have a 1650 video card with 4GB VRAM and loading the model takes 80-150 seconds each time. But something else is bad - while loading the model into video memory, the entire laptop freezes completely for these same 1-3 minutes. Is it possible to somehow change the priority of the model loading process into video memory to avoid freezes?

**Full Console Log**
c:\Fooocus>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.1.824
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 4096 MB, total RAM 16163 MB
Trying to enable lowvram mode because your GPU seems to have 4GB or less. If you don't want this use: --normalvram
Set vram state to: LOW_VRAM
Disabling smart memory management
Device: cuda:0 NVIDIA GeForce GTX 1650 : native
VAE dtype: torch.float32
Using pytorch cross attention
Refiner unloaded.
model_type EPS
adm 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra keys {'cond_stage_model.clip_l.logit_scale', 'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Base model loaded: C:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors
Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [C:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors].
Loaded LoRA [c:\Fooocus\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [C:\Fooocus\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 0.1.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cpu, use_fp16 = False.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 23524113469030253
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] palm trees against the backdrop of the Egyptian pyramids, intricate, elegant, highly detailed, very coherent, sharp focus, detail, ambient light, incredible composition, colorful, beautiful, epic, dramatic cinematic, full color, inspired, rich deep colors, amazing, futuristic, thought driven, complex, inspiring, creative, surreal, vibrant, brilliant, unique, awesome, marvelous
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding negative #1 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (1408, 704)
Preparation time: 6.10 seconds
[Sampler] refiner_swap_method = joint
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
loading in lowvram mode 1750.567232131958
[Fooocus Model Management] Moving model(s) has taken 77.29 seconds
100%|██████████████████████████████████████████████████████████████████████████████████| 30/30 [09:19<00:00, 18.63s/it]
Image generated with private log at: C:\Fooocus\Fooocus\outputs\2023-12-02\log.html
Generating and saving time: 645.32 seconds
Total time: 652.11 seconds

**Comment by mirek190** [comment]
your laptop is just too old ... 4GB VRAM and 16 GB RAM?

**Comment by Lion-kh** [comment]
Ok. Tell me please how many RAM (not VRAM) i need to work with Fooocus without page-file? 32gb? 64gb?

**Comment by Hummenix** [comment]
@Lion-kh well it's unclear. In my testing I saw as much as 16GB being used +12GB VRAM, and that is only because I only have 16GB, I am certain it would use more if it could. According to the README you need at least 8GB, WITH a page-file. 
32GB does seem reasonable, but you never know with AI unless the dev actually gives you a number. 
Also there currently seems to be a bug where the program takes more RAM than it should but as I'm on AMD I can't sufficiently isolate it to know whether it's just AMD, or in general. There are many reports of it crashing with high RAM usage on AMD if you look for it. 

I'm sorry you feel frustrated but you seem to be able to at least use the software, even if not with an ideal experience, correct me if I'm wrong there. For now I'd say to just deal with it and hope it gets better with future patches, although frankly your hardware is already on the somewhat lower end for AI work so I wouldn't get my hopes up too much.

**Comment by Lion-kh** [comment]
I am very grateful for the answer! For now the problem is conditionally resolved. I expanded the memory on my laptop to 64GB and loading the model began to take 5-9 seconds without any freezes. Based on the analysis of the use of the program, I can say that it uses about 40GB of RAM. After increasing the amount of RAM, the generation speed did not increase: one image is created in 10 minutes, but nothing can be done here.
Thanks everyone, I'm closing the request.

**Comment by lllyasviel** [comment]
this is likely a swap problem and fooocus do not need 40g ram.
see also https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md
